include/particles/particle.tcc

```diff
@@ -284,6 +290,16 @@ bool mpm::Particle<Tdim>::assign_material_state_vars(
   return status;
 }
 
+//! Assign a state variable
+template <unsigned Tdim>
+void mpm::Particle<Tdim>::assign_state_variable(const std::string& var,
```

Comment:
I think we need this to `assign_pressure` or `pore_pressure`, just to make it generalize as when we call `pressure()`, we actually calling `state_variable("pressure")`. 

I agree with you to make it in `assert` instead of throwing an error.